### PR TITLE
Catalogue tests

### PR DESCRIPTION
**Pull request by prakash87**
If you can think of any more test cases, please tell me:)

### CODE REVIEW

File: tests/app/views/test_search.py [hunk 1]
```diff
@@ -20,356 +20,89 @@ def find_search_summary(res_data):
...
-        res = self.client.get(u'/g-cloud/search?q=email+\U0001f47e&lot=saas')
-        assert_equal(200, res.status_code)
-        summary = find_search_summary(res.get_data(as_text=True))[0]
-        assert_true(
-            u'<span class="search-summary-count">1</span> result found' +
-            u' containing <em>email \U0001f47e</em> in' +
-            u' <em>Software as a Service</em>' in summary)
+class TestCataloguePage(BaseApplicationTest):
+
+    def _setUp(self, api_client, url=''):
```
Comment (sarneaud) [inline tests/app/views/test_search.py]: If you call this "setup" not "_setUp", the test framework will call it for you automatically at the start of each test.  Make sure you use "super" to call the setup method in BaseApplicationTest as well.  Look at the TestSearchSuppliers test that you deleted for an example.

File: tests/app/views/test_search.py [hunk 2]
```diff
@@ -20,356 +20,89 @@ def find_search_summary(res_data):
...
+        base_url = '/marketplace/search/suppliers'
+        url = base_url + url
+
+        self.roles_json = self._get_fixture_data('supplier_roles.json')
+        self.results_json = self._get_fixture_data('supplier_results.json')
+
+        api_client.return_value.get_roles.return_value = self.roles_json
+        api_client.return_value.find_suppliers.return_value = self.results_json
+
+        self.response = self.client.get(url)
```
Comment (sarneaud) [inline tests/app/views/test_search.py]: If you factor this request out into a method that returns a response and a page, it'll give you a more flexible interface.  It'll be easier to configure the URL, for example, or to make multiple requests in a test.

File: tests/app/views/test_search.py [hunk 3]
```diff
@@ -20,356 +20,89 @@ def find_search_summary(res_data):
...
+            for element, attribute, link, pos in result.iterlinks():
+                if element.text_content().strip() == supplier_name:
+                    found = True
+        assert found
+
+    def test_clear_buttons_have_valid_url(self, api_client):
+        self._setUp(api_client)
+        valid = True
+        for button in self.page.find_class('clear-all'):
+            if button.attrib['href'] != '/marketplace/search/suppliers':
```
Comment (sarneaud) [inline tests/app/views/test_search.py]: You import url_for at the top of the file, but don't seem to use it anywhere.  Is this where you meant to use it?

File: tests/app/views/test_search.py [hunk 4]
```diff
@@ -20,356 +20,89 @@ def find_search_summary(res_data):
...
+        self.roles_json = self._get_fixture_data('supplier_roles.json')
+        self.results_json = self._get_fixture_data('supplier_results.json')
+
+        api_client.return_value.get_roles.return_value = self.roles_json
+        api_client.return_value.find_suppliers.return_value = self.results_json
+
+        self.response = self.client.get(url)
+        self.page = html.fromstring(self.response.get_data(as_text=True))
+
+    def test_fixture_has_more_than_8_results_to_test_pagination(self, api_client):
```
Comment (sarneaud) [inline tests/app/views/test_search.py]: What's "8"?  Is it something to do with SUPPLIER_RESULTS_PER_PAGE (which is 10)?

File: tests/app/views/test_search.py (hunk 1 above)
Comment (prakash87) [inline tests/app/views/test_search.py]: Thanks for this awesome tip. Updated.

File: tests/app/views/test_search.py (hunk 2 above)
Comment (prakash87) [inline tests/app/views/test_search.py]: new method created to return response and page

File: tests/app/views/test_search.py (hunk 3 above)
Comment (prakash87) [inline tests/app/views/test_search.py]: Flask says I can't use url_for on tests. Removed the import and the hardcoded url is now placed in one single variable for easy change if required.

File: tests/app/views/test_search.py (hunk 4 above)
Comment (prakash87) [inline tests/app/views/test_search.py]: Commented it on the code

**Pull request merged**